### ISSUE TITLE
Wrong undefined variable diagnostic when using var inside callbacks used in it's declaration

### ISSUE THREAD

**Issue by georgiy-belyanin** [EmmyLuaLs/emmylua-analyzer-rust#390]
If we try to use the variable inside the expression which is used in the variable declaration the wrong diagnostics is issued even if the code works.

```lua
    local abc = watch('event', function()
        print(abc) -- abc is not defined
    end)
```

**Comment by CppCXY** [comment]
![Image](https://github.com/user-attachments/assets/f3c46ad1-9436-4731-b81f-a32931cf7232)

**Comment by georgiy-belyanin** [comment]
For some reason, I've thought this is possible. This is wrong. I am sorry. Closing it. Thanks for finding out.
